distribution/Dockerfile

```diff
@@ -0,0 +1,35 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+FROM centos:7
+
+# Project version defined in pom.xml is passed as an argument
+ARG VERSION
+
+# JDK 8 is a pre-requisite to run Drill ; 'which' package is needed for drill-config.sh
+RUN yum install -y java-1.8.0-openjdk-devel which ; yum clean all ; rm -rf /var/cache/yum
+
+# The drill tarball is generated upon building the Drill project
+COPY target/apache-drill-$VERSION.tar.gz /tmp
+
+# Drill binaries are extracted into the '/opt/drill' directory
+RUN mkdir /opt/drill
+RUN tar -xvzf /tmp/apache-drill-$VERSION.tar.gz --directory=/opt/drill --strip-components 1
+
+# Starts Drill in embedded mode and connects to Sqlline
+ENTRYPOINT /opt/drill/bin/drill-embedded
```

Comment:
We've all learned that Drill logs are essential to figure out what went wrong. As configured, these will end up in `/opt/drill/log` in the container's writeable layer. The data is lost when the container exits.

Might want to configure the container to write logs to, say, `/var/log/drill`, then encourage the user to do a bind or volume mount to that location in order to persist logs after the container exits.

Alternatively, for use in a system such as K8s, configure the logs to write to stdout so that the K8s log system can capture the log output, display it to the user (`kubectl logs <pod>`) or route it to the log aggregation system.

Reply:
Agree - this PR introduces embedded support only. But when we support stand-alone mode, either under Docker or K8S, we'll work on preserving configurations and contents. 

I have Drill working under K8S already - and the profiles have been moved to DFS (which is preserved). For logs, we'll need to figure out K8S volume plugin mounts may be. 

For a vanilla Docker env, i'll need to lookup if that's possible.

Reply:
For vanilla Docker, I suppose the user could use a Docker volume mount onto `/opt/drill/log` to capture logs... The problem is, the more directories that must be mounted this way, the messier it gets. 

As noted above, it is fine to address this in another PR.

Reply:
Sure, i'll keep this in mind.